examples/spyre_inference_ppc64le.py

```diff
@@ -0,0 +1,323 @@
+                time.sleep(1)
+        stime = time.perf_counter() - stime
+        log(f"{stime:.2f}s, connected to server")
+
+    else:
+        import vllm
+        llm = vllm.LLM(model=args.model, tensor_parallel_size=num_cards)
+        llm_sparms = vllm.SamplingParams(max_tokens=max_tokens,
+                                         temperature=0.0,
+                                         min_tokens=min_tokens)
+
+total_gentok = []
+total_time = []
+ttfirsttok = []
+lock = threading.Lock()
+
+
+async def runone(files, file_i):
+    global total_gentok
+    global total_time
+    global lock
+    global ttfirsttok
+    global tstart
+    file = files[file_i]
+    # read out the next batch_size files for prompting.
+    req_len = len(files[file_i:file_i + batch_size])
+    prompt = [
+        pathlib.Path(files[p]).read_text()
+        for p in range(file_i, file_i + req_len)
+    ]
+    comp_tok = 0
+    test_output = [""] * req_len
+    tt = 0.0
+    end_ft = 0.0
+
+    if not debug and online:
+        end_ft = -1.0
+        start = time.perf_counter()
+        try:
+            chat_completion = client.completions.create(
+                model=model,
+                prompt=prompt,
+                max_tokens=max_tokens,
+                #echo=True, streaming mode breaks with this option.
+                stream=True,
+                temperature=0.0,
+                extra_body=dict(min_tokens=min_tokens))
+            # Not sure if always true.
+            # But each iteration is one token of one request.
+            for c in chat_completion:
+                if end_ft < 0:
+                    end_ft = time.perf_counter() - start
+                comp_tok += 1
+                test_output[c.choices[0].index] += c.choices[0].text
+        except Exception:
+            tnow = time.perf_counter() - tstart
+            log(f"{tnow:.2f}s, error, prompt generation returned an exception. "
+                f"Prompt too long?")
+
+        tt = time.perf_counter() - start
+        # Check for failed prompts
+        for i in range(len(test_output)):
+            if len(test_output[i]) == 0:
+                tnow = time.perf_counter() - tstart
+                log(f"{tnow:.2f}s, error, prompt: "
+                    f"{os.path.basename(files[file_i+i])}"
+                    f" did not generate any tokens. prompt likely too long")
+        test_output = [prompt[i] + test_output[i] for i in range(req_len)]
+    elif not debug and not online:
+        start = time.perf_counter()
+        output = llm.generate(prompts=prompt,
+                              sampling_params=llm_sparms,
+                              use_tqdm=False)
+        tt = time.perf_counter() - start
+
+        test_output = [
+            prompt[i] + output[i].outputs[0].text for i in range(len(output))
+        ]
+
+        # From the innards of vllm, hopefully accurate
+        end_ft = output[0].metrics.first_token_time - output[
```

Comment:
Do these metrics work with vllm v1 as well? I think they were dropped from some of the python apis during the move